### ISSUE TITLE
return in finally can swallow exceptions

### ISSUE THREAD

**Issue by iritkatriel** [couchbase/couchbase-python-client#57]
In https://github.com/couchbase/couchbase-python-client/blob/465027c0253e3ba059eb1045793e5a3689bfa00f/acouchbase/transactions/transactions.py#L85

there is this snippet:
```
                try:
                    fn(self, *args, **kwargs)
                except SystemError as e:
                    ftr.set_exception(e.__cause__)
                except Exception as e:
                    ftr.set_exception(e)
                finally:
                    return ftr
```

It has a `return` statement in a `finally` block, which would swallow any in-flight exception.

This means that if a `BaseException` (such as `KeyboardInterrupt`) is raised from the `try` body, or any exception is raised from an `except`: clause, it will not propagate on as expected.

See also https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions.

**Comment by thejcfactor** [comment]
Hi @iritkatriel -- Thank you for filing the issue.  I have created [PYCBC-1636](https://jira.issues.couchbase.com/browse/PYCBC-1636) in our JIRA system and will look to make changes as soon as possible.

**Comment by thejcfactor** [comment]
Hi @iritkatriel -- [PYCBC-1636](https://jira.issues.couchbase.com/browse/PYCBC-1636) has been resolved and [v4.3.4](https://pypi.org/project/couchbase/) has been released which includes the fix for PYCBC-1636.
